public/components/metrics/sidebar/metric_name.tsx

```diff
@@ -25,15 +30,16 @@ interface IMetricNameProps {
 export const MetricName = (props: IMetricNameProps) => {
   const { metric, handleClick } = props;
 
-  const name = () => {
-    if (metric.catalog === 'CUSTOM_METRICS') return metric.name;
-    else return metric.name.split('.')[1].replace(/^prometheus_/, 'p.._');
+  const name = (metricDetails: any) => {
+    if (metricDetails?.catalog === 'CUSTOM_METRICS' || metricDetails?.catalog === 'OpenTelemetry')
```

Comment:
More logic into metricSlice.   Think about ways to remove if-conditions.  Branching creates what we call "cyclomatic complexity".  The complexity increases exponentially on every branch.   A function with 2 conditional switches is 4!! times harder to understand!

Reply:
should we add cyclomatic complexity in lint? i remember eslint supports it https://eslint.org/docs/latest/rules/complexity

Reply:
modified it